tool_path_planner/src/halfedge_boundary_finder.cpp

```diff
@@ -140,6 +142,191 @@ bool decimate(const pcl::PointCloud<pcl::PointNormal>& in, pcl::PointCloud<pcl::
    return out.size() > 2;
 }
 
+/**
+ * @brief forces the required point spacing by computing new points along the curve
+ * @param in    The input cloud
+ * @param out   The output cloud
+ * @param dist  The required point spacing distance
+ * @return True on success, False otherwise
+ */
+bool applyEqualDistance(const pcl::PointCloud<pcl::PointNormal>& in, pcl::PointCloud<pcl::PointNormal>& out,double dist)
+{
+  using namespace pcl;
+  using namespace Eigen;
+
+  // kd tree will be used to recover the normals
+  PointCloud<PointXYZ>::Ptr in_points = boost::make_shared< PointCloud<PointXYZ> >();
+  copyPointCloud(in, *in_points);
+  pcl::KdTreeFLANN<pcl::PointXYZ> kdtree;
+  kdtree.setInputCloud(in_points);
+  kdtree.setEpsilon(EPSILON);
+
+  std::remove_reference< decltype(in) >::type::PointType new_pn;
+  PointXYZ p_start, p_mid, p_end;
+  Vector3f v_1 ,v_2, unitv_2, v_next;
+  p_start = (*in_points)[0];
+  p_mid = (*in_points)[1];
+  v_1 = p_mid.getVector3fMap() - p_start.getVector3fMap();
+  out.push_back(in[0]); // adding first points
+  for(std::size_t i = 2; i < in_points->size(); i++)
+  {
+    p_end = (*in_points)[i];
+    while(dist < v_1.norm())
+    {
+      p_start.getVector3fMap() = p_start.getVector3fMap() + dist * v_1.normalized();
+      std::tie(new_pn.x, new_pn.y, new_pn.z) = std::make_tuple(p_start.x, p_start.y, p_start.z);
+      out.push_back(new_pn);
+      v_1 = p_mid.getVector3fMap() - p_start.getVector3fMap();
+    }
+
+    v_2 = p_end.getVector3fMap() - p_mid.getVector3fMap();
+    if(dist < (v_1 + v_2).norm())
+    {
+      // solve for x such that " x^2 + 2 * x * dot(v_1, unitv_2) + norm(v_1)^2 - d^2 "
```

Comment:
Fix comment by adding an equal zero or similar to make sense x^2 +2x*v_1*uv2 + ||v1||^2 - d^2 = 0